libs/designer/src/lib/ui/configuretemplate/workflows/selectWorkflows.tsx

```diff
@@ -50,10 +50,12 @@ export const SelectWorkflows = ({
         const normalizedWorkflowId = id.toLowerCase();
         onWorkflowsSelected([normalizedWorkflowId]);
       } else {
-        onWorkflowsSelected([]);
+        const workflowsOfSameSubscription =
+          Object.values(selectedWorkflowsList).map((workflow) => workflow.manifest?.metadata?.workflowSourceId as string) ?? [];
```

Comment:
Not able to understand this part, why are we saving the same workflows list when logicapp is reselected... should you be checking if they belong to the same parent here before getting the current list